src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -196,6 +198,138 @@ public void GenerateHeader()
             PopBlock();
         }
 
+        private class DelegateSignature
+        {
+            public string retType, signature;
```

Comment:
As per Mono/.NET conventions, please capitalise public members. Also, as I mentioned once, public fields are bad, properties should be used instead. As I additionally mentioned then, we do have public fields in other places but that doesn't mean it's right, and we've been replacing them over time.